### PR TITLE
Add Visp recipe

### PR DESCRIPTION
**Pull request by olivier-roussel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fspindle** [comment]
Thanks @olivier-roussel 
As lead developer of ViSP I confirm that I'm willing to be listed in the maintainer section.

**Comment by olivier-roussel** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/visp/build.sh
Comment (jjerphan) [inline recipes/visp/build.sh:38]: Must it be passed?
```suggestion
      -DCMAKE_BUILD_TYPE=Release \
      -DOGRE_DIR=${OGRE_DIR} \
```

**Review by jjerphan — COMMENTED** [review]
Thank you, @olivier-roussel.

This recipe is nicely documented.

I have a few minor suggestions and questions.

File: recipes/visp/meta.yaml
Comment (olivier-roussel) [reply recipes/visp/meta.yaml]: Agreed. Done.

File: recipes/visp/build.sh
Comment (olivier-roussel) [reply recipes/visp/build.sh:38]: Here I have used the `OGRE_DIR` environment variable, but I could also have used the `OGRE_DIR` cmake variable passed to the `cmake` command as you mentioned, as both are checked in the `CMakeLists.txt`. 
The only motivation behind using the environment variable option here was to keep some code readability due to the specialization between linux and osx. But if the use of a cmake variable should be preferred, I can switch to it, using a shell variable for clarity for example.

File: recipes/visp/meta.yaml
Comment (olivier-roussel) [reply recipes/visp/meta.yaml:60]: Indeed, good to know. I'll make the change.

**Comment by olivier-roussel** [comment]
@jjerphan I've added your remarks to the PR

File: recipes/visp/bld.bat
Comment (jjerphan) [inline recipes/visp/bld.bat:12]: For consistency?
```suggestion
    %CMAKE_ARGS% ^
    %SRC_DIR% ^
```

File: recipes/visp/bld.bat
Comment (jjerphan) [inline recipes/visp/bld.bat:14]: Could you please:
 - define a variable for `OGRE_DIR` similarly to what's done for Unix
 - document the reason for this change, referring to the patch

?

File: recipes/visp/bld.bat
Comment (jjerphan) [inline recipes/visp/bld.bat:21]: Nit: Grouping Windows specific-argument together (I do not know if one can put a comment to separate both groups of arguments without breaking the command line).
```suggestion
    -G Ninja ^
    -DBUILD_TESTS=ON ^
    -DOGRE_DIR="%LIBRARY_PREFIX%\cmake" ^
    -DCMAKE_BUILD_TYPE:STRING=Release ^
    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
    -DVISP_LIB_INSTALL_PATH:PATH="lib" ^
    -DVISP_BIN_INSTALL_PATH:PATH="bin" ^
    -DVISP_CONFIG_INSTALL_PATH:PATH="cmake" ^
    -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX%
```

File: recipes/visp/meta.yaml
Comment (jjerphan) [inline recipes/visp/meta.yaml:33]: Nit: Yes, I am curing my OCD.
```suggestion
    - make                           # [not win]
    - clang                          # [win]
    - ninja                          # [win]
    - {{ cdt('mesa-libgl-devel') }}  # [linux]
    - {{ cdt('mesa-dri-drivers') }}  # [linux]
    - {{ cdt('libselinux') }}        # [linux]
    - {{ cdt('libxdamage') }}        # [linux]
    - {{ cdt('libxxf86vm') }}        # [linux]
    - {{ cdt('libxext') }}           # [linux]
```

File: recipes/visp/meta.yaml
Comment (jjerphan) [inline recipes/visp/meta.yaml:44]: Nit: I am still quite not mad.
```suggestion
    - xorg-libx11                   # [unix]
    - xorg-libxfixes                # [unix]
    - libxml2                       # [unix]
    - libdc1394 >=2.2.6             # [unix]
    - librealsense                  # [unix]
    - _openmp_mutex                 # [linux]
    - libopenblas                   # [osx]
    - llvm-openmp                   # [win or osx]
    - openmp                        # [win]
    - pthreads-win32                # [win]
```

File: recipes/visp/meta.yaml
Comment (jjerphan) [inline recipes/visp/meta.yaml:14]: Could you document the reason to disable those instruction sets on Windows?

File: recipes/visp/meta.yaml
Comment (jjerphan) [inline recipes/visp/meta.yaml:64]: Nit: and I'll stop there. 
```suggestion
    - test -f $PREFIX/include/visp3/visp.h                                 # [unix]
    - test -f $PREFIX/lib/cmake/visp/VISPConfig.cmake                      # [unix]
    - test -f $PREFIX/lib/cmake/visp/VISPConfig-version.cmake              # [unix]
    - test -f $PREFIX/lib/lib{{ name }}_core${SHLIB_EXT}                   # [unix]
    - if not exist %LIBRARY_PREFIX%\include\visp3\visp.h exit 1            # [win]
    - if not exist %LIBRARY_PREFIX%\cmake\VISPConfig.cmake exit 1          # [win]
    - if not exist %LIBRARY_PREFIX%\cmake\VISPConfig-version.cmake exit 1  # [win]
```

**Review by jjerphan — APPROVED** [review]
Thanks for the heads-up, here are some last comments.

File: recipes/visp/meta.yaml
Comment (olivier-roussel) [reply recipes/visp/meta.yaml:33]: Agreed, this is much more readable. I thought linting rules were fixed to two spaces before he platform specification, but this seems OK for the linter. Will do.

File: recipes/visp/meta.yaml
Comment (jjerphan) [reply recipes/visp/meta.yaml:17]: I don't have the final word on what must be published on conda-forge but deactivating it while documenting the reason is reasonable and sufficient to me; this can be fixed with other builds or with another version.

Comment (olivier-roussel) [reply recipes/visp/meta.yaml:21]: Fine. Thanks for your feedback.

**Comment by olivier-roussel** [comment]
Hi @jjerphan, I have fixed your remarks. Thanks a lot for your valuable review, it will also help for next recipes to come.

File: recipes/visp/meta.yaml
Comment (carterbox) [inline recipes/visp/meta.yaml:50]: https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

Here's our instructions on how to build with openmp enabled. I'm not sure why, but we recommend to put these deps in the build section.

File: recipes/visp/meta.yaml
Comment (carterbox) [inline recipes/visp/meta.yaml:65]:
```suggestion
  commands:
    - visp-config --version
    - test -f $PREFIX/include/visp3/visp.h                                                   # [unix]
```

**Review by carterbox — CHANGES_REQUESTED** [review]
This is a very polished recipe. Good job! Just some minor suggestions.

**Comment by olivier-roussel** [comment]
@carterbox Thanks for your feedback ! It seems that 2d48099 broke the Windows build, even if I don't see why yet. Will fix and do suggested changes on the feedstock.

**Pull request merged**
